Promise.lib.nut

```diff
@@ -240,21 +239,30 @@ class Promise {
     // Make every promise in array of promises not last
     // for suppressing Unhadled Exeptions Warnings
     //
+    // FROM 4.0.1 -- Update function to avoid calling 'array[any exp.]()' as
+    //               identified by @gjanvier, and to check for non-array params
+    //
     // @param {{Promise|function}[]} promises - array of Promises/functions that return Promises
     // @return {Promise} Promise that is resolved/rejected with the last value that come from looped promise
     //
     static function serial(promises) {
+        if (typeof promises != "array") {
+            return Promise.reject("Promise.serial() not passed an array");
+        }
+
         local i = 0;
         return loop(
-            @() i < promises.len(),
+            @() i < promises.len(),     // This is a Squirrel lamda --
+                                        // see https://developer.electricimp.com/squirrel/squirrel-guide/functions#lambda-functions
             function () {
-                return "function" == type(promises[i])
-                     ? promises[i++]()
-                     : promises[i++];
+                local res = promises[i++];
```

Comment:
Suggest a comment right here explaining the problem, so that no-one's tempted to refactor it back to the broken behaviour.